products/monitoring/api.yaml

```diff
@@ -0,0 +1,785 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+--- !ruby/object:Api::Product
+name: Stackdriver Monitoring and Alerting
+prefix: gmonitoring
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://monitoring.googleapis.com/v3/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+objects:
+  - !ruby/object:Api::Resource
+    name: 'AlertPolicy'
+    base_url: projects/{{project}}/alertPolicies
+    self_link: projects/{{project}}/alertPolicies/{{name}}
+    update_verb: :PATCH
+    input: true
+    description: |
+      A description of the conditions under which some aspect of your system is
+      considered to be "unhealthy" and the ways to notify people or services
+      about this state.
+    exports:
+      - name
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/monitoring/docs/'
+      api: 'https://cloud.google.com/monitoring/api/ref_v3/rest/'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          The resource name for this policy.
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+         A short name or phrase used to identify the policy in
+         dashboards, notifications, and incidents. To avoid confusion, don't use
+         the same display name for multiple policies in the same project. The
+         name is limited to 512 Unicode characters.
+        required: true
+      - !ruby/object:Api::Type::Enum
+        name: combiner
+        description: |
+          How to combine the results of multiple conditions to
+          determine if an incident should be opened.
+        values:
+          - :AND
+          - :OR
+          - :AND_WITH_MATCHING_RESOURCE
+        required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: creationRecord
+        description: |
+          A read-only record of the creation of the alerting policy.
+          If provided in a call to create or update, this field will
+          be ignored.
+        output: true
+        properties:
+         - !ruby/object:Api::Type::String
+           name: mutateTime
+           description: |
+             When the change occurred.
+         - !ruby/object:Api::Type::String
+           name: mutatedBy
+           description: |
+             The email address of the user making the change.
+      - !ruby/object:Api::Type::Boolean
+        name: enabled
+        # TODO fix MM to accept default boolean values
```

Comment:
It was more difficult then I thought it would be. It's some [meta logic in type.rb](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/api/type.rb#L76) but because ruby doesn't actually have a normal 'boolean' class and instead has native classes of TrueClass and FalseClass it will take a little bit of work. I'll give it a shot though and see if I can get it in.

Reply:
❓ should this be `send_empty_value: true` so that somebody can send enabled false?

Reply:
probably, but the easiest way to find out is going to be to try it out and see what happens